### PR TITLE
Himanshi/create role UI

### PR DESCRIPTION
**Pull request by himanshi-chhabra**
### :nut_and_bolt: Description: What code changed, and why?
We need to add UI for Create Role

### :chains: Related Resources
#4544 

### :+1: Definition of Done
Added create functionality for Role in which we can create a particular Role.
### :athletic_shoe: How to Build and Test the Change
```
STEP 1
inside the hab studio

[default:/src:0]# build components/automate-ui-devproxy/
[default:/src:0]# start_automate_ui_background
[default:/src:0]# start_all_services

STEP 2
open new window
go to automate UI path

$ cd components/automate-ui
and run the command 

npm run serve:hab
```
To add data https://github.com/chef/automate/blob/master/dev-docs/adding-data/adding_test_data.md#adding-data-to-infra-views
 **PFA video for these changes.**

Steps: 
 1. Go To  Infrastructure -> Chef Servers -> click server name-> click on org name -> Role
 2. Click create role button

### :white_check_mark: Checklist

- [ ] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] Tests added/updated?
- [ ] Docs added/updated?
- [ ] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### :camera: Screenshots, if applicable
create_role_1
create_role_2
create_role_3
create_role_4
create_role_5

https://user-images.githubusercontent.com/61003053/111340826-fc40ec00-869e-11eb-939b-dcc94c6fe4ca.mp4

### CODE REVIEW

File: components/automate-ui/src/app/modules/infra-proxy/infra-role-details/infra-role-details.component.ts [hunk 1]
```diff
@@ -106,7 +106,7 @@ export class InfraRoleDetailsComponent implements OnInit, OnDestroy {
       this.expandedList = role.expanded_run_list;
       this.runList = this.role.run_list;
       this.idList = [];
-      this.attributes = new RoleAttributes(this.role);
+      // this.attributes = new RoleAttributes(this.role);
```
Comment (shaik80) [inline components/automate-ui/src/app/modules/infra-proxy/infra-role-details/infra-role-details.component.ts:109]: why we are commenting this, can u explain

**Review by shaik80 — COMMENTED** [review]
https://user-images.githubusercontent.com/47217471/111768998-abafd580-88ce-11eb-9597-9c630f1fb8f8.mov

how is this up arrow and down arrow works
It works only in ascending  and descending  order,
or it gives u the flexibility to move list data one step up and one step down..?
can explain this functionality..?

**Comment by himanshi-chhabra** [comment]
>  Screen.Recording.2021-03-18.at.9.30.22.PM.mov 
> how is this up arrow and down arrow works
> It works only in ascending and descending order,
> or it gives u the flexibility to move list data one step up and one step down..?
> can explain this functionality..?

@shaik80 these up and down arrows will sort the dropped data into ascending and descending order.

File: components/automate-ui/src/app/modules/infra-proxy/infra-role-details/infra-role-details.component.ts (hunk 1 above)
Comment (himanshi-chhabra) [reply components/automate-ui/src/app/modules/infra-proxy/infra-role-details/infra-role-details.component.ts]: i have updated the changes

File: components/automate-ui/src/app/modules/infra-proxy/drag-drop/drag-drop.component.html [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            [list]="availableType"
+            search="true" sort="true" selectAll="true"
+            [selectedItems]="selectedItems"
+            (selectedValues)="selectedItemsHandler($event)">
+            </app-select-box>
+            <br />
+        </form>
+        <br />
+    </div>
+</div>
```
Comment (vinay033) [inline components/automate-ui/src/app/modules/infra-proxy/drag-drop/drag-drop.component.html:36]: add one line here, also check the indentation once.

**Review by vinay033 — APPROVED** [review]
changes look good to me, need to fix some minor things.

**Comment by shaik80** [comment]
> > Screen.Recording.2021-03-18.at.9.30.22.PM.mov
> > how is this up arrow and down arrow works
> > It works only in ascending and descending order,
> > or it gives u the flexibility to move list data one step up and one step down..?
> > can explain this functionality..?
> 
> @shaik80 these up and down arrows will sort the dropped data into ascending and descending order.

**Review by shaik80 — APPROVED** [review]
Everything looks good to me,
Except for the **current run list** block, if we select multiple data from the list and then press the up or down arrow from the UI,
It gives me the result in ascending or descending order, If we are sorting in ascending or descending order, then what is the point in selecting multiple data from the list..?
If we try to think from a customer's viewpoint it will be confusing to them
please try to create an issue ticket for fixing this.

**Comment by himanshi-chhabra** [comment]
> Everything looks good to me,
> Except for the **current run list** block, if we select multiple data from the list and then press the up or down arrow from the UI,
> It gives me the result in ascending or descending order, If we are sorting in ascending or descending order, then what is the point in selecting multiple data from the list..?
> If we try to think from a customer's viewpoint it will be confusing to them
> please try to create an issue ticket for fixing this.

@shaik80 Agree with you, discussed this with UX team will take these changes in this issue https://github.com/chef/automate/issues/4540

**Comment by himanshi-chhabra** [comment]
While acceptance testing, observed
- When we type on the text field the borderline (stroke) should be in our primary blue color.
- When we select Roles of Recipes in the Run list view, there should be a hover value.
- When we select roles and recipes to drop in the Current run list. The selection is a little offline
- No hover value for arrows, and search bar.
- Disable UP and down arrows if item not selected
- Space between the Current run list and buttons(left/right).
- search bar border is overlapping by padding.
- width of dropdown and available roles and recipes not the same.
will cover in #4540

**Pull request merged**